### PR TITLE
journalpump: inject __SEQNUM into journal entries

### PR DESCRIPTION
**Pull request by ezotrank**
To get the log sequence number, to use it after in sort order.

Parse the sequence number from the i= field of the journald cursor string and include it as an integer under __SEQNUM in each entry returned by PumpReader.get_next(). This allows downstream consumers to use it as a secondary sort key for log lines sharing the same microsecond timestamp (e.g. MySQL crash stack traces).

No direct API exists on the systemd.journal.Reader C extension for sequence numbers, so the cursor string is the only available source, consistent with how __REALTIME_TIMESTAMP is injected via _get_realtime().

### CODE REVIEW

File: journalpump/journalpump.py [hunk 1]
```diff
@@ -102,14 +102,26 @@ def convert_entry(self, entry):
 
         return output
 
+    @staticmethod
+    def _parse_seqnum_from_cursor(cursor):
+        # cursor format: s=...;i=<hex_seqnum>;b=...;m=...;t=...;x=...
+        for part in cursor.split(";"):
+            if part.startswith("i="):
+                return int(part[2:], 16)
```
Comment (dogukancagatay) [inline journalpump/journalpump.py]: We should handle the `ValueError` `int()` may possibly throw.

File: test/test_journalpump.py [hunk 2]
```diff
@@ -261,6 +262,25 @@ def test_journalpump_init(tmpdir):  # pylint: disable=too-many-statements
             assert isinstance(s, GoogleCloudLoggingSender)
 
 
+def test_pump_reader_get_next_includes_seqnum():
+    reader = PumpReader()
+    cursor = "s=abc;i=1f4;b=def;m=123;t=456;x=789"
```
Comment (dogukancagatay) [inline test/test_journalpump.py:267]: We should add tests for the edge cases. e.g. no seqnum field, empty seqnum, etc.

File: test/test_journalpump.py [hunk 3]
```diff
@@ -261,6 +262,25 @@ def test_journalpump_init(tmpdir):  # pylint: disable=too-many-statements
...
+        mock.patch.object(journal.Reader, "_next", return_value=True),
+        mock.patch.object(journal.Reader, "_get_all", return_value=entry_data),
+        mock.patch.object(reader, "_get_realtime", return_value=1000000),
+        mock.patch.object(reader, "_get_cursor", return_value=cursor),
+    ):
+        result = reader.get_next()
+
+    assert result is not None
+    assert "__SEQNUM" in result.entry
+    assert result.entry["__SEQNUM"] == 0x1F4
```
Comment (dogukancagatay) [inline test/test_journalpump.py]: nit: could use named constant to be more descriptive.

File: journalpump/journalpump.py [hunk 4]
```diff
@@ -102,14 +102,26 @@ def convert_entry(self, entry):
 
         return output
 
+    @staticmethod
+    def _parse_seqnum_from_cursor(cursor):
```
Comment (dogukancagatay) [inline journalpump/journalpump.py]: Missing return type hint.
```suggestion
    def _parse_seqnum_from_cursor(cursor) -> int | None:
```

**Review by dogukancagatay — APPROVED** [review]
LGTM
